### PR TITLE
Refactor Snippets delete view to extend from generic DeleteView

### PR DESCRIPTION
**Pull request by laymonage**
Follow up to #8332, #8335, and #8344.

_Please check the following:_

- [x] Do the tests still pass?[^1]
- [x] Does the code comply with the style guide? 
    - [x] Run `make lint` from the Wagtail root. 
- [ ] For Python changes: Have you added tests to cover the new/fixed behaviour?
- [ ] For front-end changes: Did you test on all of Wagtail’s supported environments?[^2]
    - [ ] **Please list the exact browser and operating system versions you tested**:
    - [ ] **Please list which assistive technologies [^3] you tested**: 
- [ ] For new features: Has the documentation been updated accordingly?

**Please describe additional details for testing this change**. 

[^1]: [Development Testing](https://docs.wagtail.org/en/latest/contributing/developing.html#testing)
[^2]: [Browser and device support](https://docs.wagtail.org/en/latest/contributing/developing.html#browser-and-device-support)
[^3]: [Accessibility Target](https://docs.wagtail.org/en/latest/contributing/developing.html#accessibility-targets)

### CODE REVIEW

File: wagtail/snippets/views/snippets.py [hunk 1]
```diff
@@ -494,6 +494,7 @@ def setup(self, request, *args, **kwargs):
         self.model_name = kwargs.get("model_name")
         self.pk = kwargs.get("pk")
         self.model = self._get_model()
+        self.object = self.get_object()
```
Comment (kaedroho) [inline wagtail/snippets/views/snippets.py:513]: Could we call this ``get_objects``/``objects`` to indicate it's a list? It also might be nice to decide whether to use ``objects`` or ``instances`` in both the view and template for consistency.

File: wagtail/snippets/views/snippets.py [hunk 2]
```diff
@@ -578,11 +578,10 @@ def delete_action(self):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        count = len(self.object)
         context.update(
             {
                 "model_opts": self.model._meta,
-                "count": count,
+                "count": len(self.object),
```
Comment (kaedroho) [inline wagtail/snippets/views/snippets.py:584]: We could do this in the template with ``{{ instances|length }}``

**Review by kaedroho — CHANGES_REQUESTED** [review]
Nice work! Only a few minor comments

File: wagtail/snippets/views/snippets.py [hunk 3]
```diff
@@ -487,15 +487,22 @@ def edit(request, app_label, model_name, pk):
 
 
 class Delete(DeleteView):
-    def get(self, request, *args, app_label, model_name, pk=None, **kwargs):
-        model = get_snippet_model_from_url_params(app_label, model_name)
+    def setup(self, request, *args, **kwargs):
```
[the remark this answers is not in the document]
Comment (laymonage) [reply wagtail/snippets/views/snippets.py:506]: Okay, do you want the same change applied to the previously merged views in this PR as well?

File: wagtail/snippets/views/snippets.py (hunk 1 above)
Comment (laymonage) [reply wagtail/snippets/views/snippets.py:513]: Yeah but that means we'll need to override `post()` and `form_valid()` as well:

https://github.com/wagtail/wagtail/blob/499750735962c2a0f2b5abd32eaf8e6359490a57/wagtail/admin/views/generic/models.py#L38-L53

Would that be okay?

File: wagtail/snippets/views/snippets.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaedroho) [reply wagtail/snippets/views/snippets.py:506]: Yep, sorry for not spotting this earlier. Happy for this to be made in a separate PR

**Review by kaedroho — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**